web/client/src/models/file.ts

```diff
@@ -53,4 +56,9 @@ export class ModelFile extends ModelArtifact<InitialFile> {
   get isSQLMeshSeed(): boolean {
     return ['.csv'].includes(this.extension) && this.path.startsWith('seeds/')
   }
+
+  updateContent(newContent: string = ''): void {
```

Comment:
Is content updated when user types into the editor?

Reply:
yes, but we also need to reset `_content` when we saving file content to keep track of changed state